src/components/Tools/TwoDDriveSim.tsx

```diff
@@ -0,0 +1,208 @@
+    }
+
+    function findIntersection(lEnd, rEnd, context)
+    {
+        let a = {x : lPosX + wheelWidth/2, y : lPosY + wheelHeight/2};
+        let b = {x : lEnd.x, y : lEnd.y};
+        let c = {x : rPosX + wheelWidth/2, y : rPosY + wheelHeight/2};
+        let d = {x : rEnd.x, y : rEnd.y};
+
+        // console.log('a == ', 'x:', a.x, 'y:', a.y);
+        // console.log('b == ', 'x:', b.x, 'y:', b.y);
+        // console.log('c == ', 'x:', c.x, 'y:', c.y);
+        // console.log('d == ', 'x:', d.x, 'y:', d.y);
+
+        let m = (d.x - c.x) * (a.y - c.y) - (d.y - c.y) * (a.x - c.x);
+        let n = (d.y - c.y) * (b.x - a.x) - (d.x - c.x) * (b.y - a.y);
+        
+        let u =  m / n;
+
+        let x = a.x + u * (b.x - a.x);
+        let y = a.y + u * (b.y - a.y);
+
+        //console.log('intersection = x: ' , x.toFixed(4), ' y: ', y.toFixed(4));
+        //draw turning radius position dot
+        // context.save();
+        //     context.beginPath();
+        //     context.arc(x, y, 5, 0, 2 *Math.PI);
+        //     context.stroke();
+        // context.restore();
+        return {x : x.toFixed(4), y : y.toFixed(4)};
+    }
+
+    function drawPath(context, wheel, posX, posY, angle, intersection, color){
+        let radius = Math.sqrt(Math.pow(intersection.x - posX, 2) + Math.pow(intersection.y - posY, 2));
+        let radian = angle * Math.PI/180;
+        context.save();
+            context.translate(intersection.x, intersection.y);
+            context.lineWidth = 5;
+            context.strokeStyle = color;
+            if(rover.left == rover.right && rover.right == rover.back)
+            {
+                context.translate(-intersection.x, -intersection.y);
+                context.translate(posX, posY)
+                context.rotate(angle * Math.PI/180)
+                context.translate(-posX, -posY)
+
+                context.beginPath();
+                context.moveTo(posX, posY)
+                context.lineTo(posX, -1000);
+                context.stroke();
+            }
+            else if(angle > 0 && wheel != 'back')
+            {
+                context.rotate(angle * (Math.PI/180));
+                context.beginPath();
+                //displays radius
+                // context.arc(0, 0, 5, 0, 2 *Math.PI);
+                // context.stroke();
+                context.arc(0, 0, Math.abs(radius),Math.PI, (1.5 * Math.PI), 0);
+                context.stroke();
+            }
+            else if(angle < 0 && wheel != 'back')
+            {
+                context.rotate(angle * (Math.PI/180));
+                context.beginPath();
+                //displays radius
+                // context.arc(0, 0, 5, 0, 2 *Math.PI);
+                // context.stroke();
+                context.arc(0, 0, Math.abs(radius),0, (1.5 * Math.PI), 1);
+                context.stroke()
+            }
+            
+        context.restore();
+    }
+
+    function drawGrid(context, cWidth, cHeight, pad)
```

Comment:
The grid is a little noisy visually, we could make it lighter or less opaque to better emphasize the wheel display